manifests/init.pp

```diff
@@ -1,60 +1,424 @@
+#   The path to the service executable for the sssd service
+#   Default value: '/etc/init.d/sssd'
+#
+# @param sssd_service_name [Optional[String]]
+#   The name of the sssd service
+#   Default value: 'sssd'
+#
+# @param tools_package_name [Optional[String]]
+#   The name of the sssd tools package
+#   Default value: 'sssd-tools'
+#
+# @param tools_package_version [Optional[String]]
+#   The version of the tools package to be installed
+#   Default value: 'latest'
+#
+# === SIMP Integration Parameters ===
+# These parameters control the integration of the sssd module with
+# other SIMP modules and services. The variables that are being set here are
+# resolved using simplib::lookup, a special function that helps SIMP
+# integrate existing modules. Setting these values in hiera will supersede
+# simplib::lookup, and is not recommended.
+#
+# @param app_pki_dir [Stdlib::Absolutepath]
+#   Controls the basepath of the $app_pki_key, $app_pki_cert,
+#   $app_pki_ca, $app_pki_ca_dir, and $app_pki_crl parameters.
+#   Default value: '/etc/pki/simp_apps/sssd/x509'
+#
+# @param auditd [Boolean]
+#   Determines if auditd is enabled at a global level. If it is,
+#   this module will take steps to ensure sssd and auditd work
+#   together.
+#   Default value: false
+#
+# @param pki [[Boolean,Enum['simp']]]
+#   Determines if pki is enabled at a global level. If it is,
+#   this module will take steps to ensure that sssd and pki work
+#   together.
+#   Default value: false
+#
+# @param app_pki_cert_source [Stdlib::Absolutepath]
+#   Determines the location that pki will look for certificates.
+#   Default value: '/etc/pki/simp/x509'
+#
+# === Template Parameters ===
+# The following parameters are exposed here, but are only used inside of
+# templates that configure the sections of the sssd configuration file.
+# Futher information about which parameters are available to each section
+# of the configuration file can be found inside the manifests that control
+# the corresponding sections. (ex: manifests/service/pam.pp)
+#
+# @param allowed_uids [Array[String]]
+#   Sets the allowed_uids field in the pac section of the sssd config file.
+#   Default value: []
+#   Used in: templates/service/pac.erb
+#
+# @param autofs_negative_timeout [Optional[Integer]]
+#   Sets autofs_negative_timeout in the autofs section of the sssd config file
+#   Default value: null
+#   Used in: templates/service/autofs.erb
+#
+# @param command [Optional[String]]
+#   Sets command in the nss and pam sections of the sssd config file
+#   Default value: null
+#   Used in: templates/service/nss.erb
+#            templates/service/pam.erb
+#
+# @param config_file_version [Integer]
+#   Sets config_file_version in the sssd section of the sssd config file
+#   Default value: 2
+#   Used in: templates/sssd.conf.erb
+#
+# @param debug_level [Optional[Sssd::DebugLevel]]
+#   Sets debug_level in each section of the config file.
+#   Default value: null
+#   Used in: templates/domain.erb
```

Comment:
Regarding Puppet-Strings, Dylan says there is a major update in progress, and that it would be prudent to wait and see what functionality is added. Apparently support for the new lookup will be part of it.